pkg/internal/codegen/parse/util.go

```diff
@@ -326,3 +340,21 @@ func checkType(props *v1beta1.JSONSchemaProps, s string, enums *[]v1beta1.JSON)
 		*enums = append(*enums, v1beta1.JSON{Raw: []byte(`"` + s + `"`)})
 	}
 }
+
+// Scale subresource requires specpath, statuspath, selectorpath key values, represents for JSONPath of
+// SpecReplicasPath, StatusReplicasPath, LabelSelectorPath separately. e.g.
+// +kubebuilder:subresource:scale:specpath=.spec.replica,statuspath=.status.replica,selectorpath=
+func parseJSONPath(t *types.Type) map[string]string {
+	jsonPath := make(map[string]string)
+	for _, c := range t.CommentLines {
+		if strings.Contains(c, "+kubebuilder:subresource:scale") {
+			paths := strings.Split(c, ":")[3:][0]
+			path := strings.Split(paths, ",")
+			for _, s := range path {
+				kv := strings.Split(s, "=")
+				jsonPath[kv[0]] = kv[1]
```

Comment:
Lets be conservative and do all sort of checking here. For ex.
len(kv) > 1
k has to be one of ("specpath", "statuspath", "selectorpath")
return an error with good text with recommended fix.

Reply:
Good point.